### PR TITLE
Make wc provider into its own package

### PR DESCRIPTION
**Pull request by Aierie**
CS-2897

Had some difficulty handling the typescript declarations/module overwrites' interactions with webpack, so there are some new `// @ts-ignores` 😭 , but it should be working for now both in the hub and in web-client.

We could merge and deploy the stopgap PR (https://github.com/cardstack/cardstack/pull/2543) if this might take some time?

TODO:
- [x] restore socket disconnect events (used in web client to inform the user that they are disconnected and need to refresh the page)
- [x] add the logging of disconnection/connection of websockets (browser and Sentry) back to web client
- [x] trim unused types and dependencies
- [x] review tests of WalletConnectProvider to see if they are applicable here, and port them over
- [x] License update
- [ ] better documentation

### CODE REVIEW

**Comment by Aierie** [comment]
https://linear.app/cardstack/issue/CS-2897/investigate-call-volume-from-web-app

**Comment by Aierie** [comment]
Tried porting tests over , but it wasn't too easy to get things working with a websocket provider.

File: packages/wc-provider/package.json [hunk 1]
```diff
@@ -0,0 +1,38 @@
+{
+  "name": "@cardstack/wc-provider",
+  "version": "0.28.15",
+  "private": true,
+  "description": "WalletConnect provider that uses websockets rather than http",
+  "author": "Aierie <michael.khor@cardstack.com>",
+  "homepage": "https://github.com/cardstack/cardstack#readme",
+  "license": "MIT",
```
Comment (habdelra) [inline packages/wc-provider/package.json:8]: I think this is actually a GPL license (this PR actually solves another issue that we had around separating the WC-provider into it's own package. @lukemelia @ef4 do you guys know of any additional things we need to do in order to indicate this is a GPL license?

File: types/web3-provider-engine/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
+declare module 'web3-provider-engine' {
```
Comment (habdelra) [inline types/web3-provider-engine/index.d.ts:1]: i think this was accidentally committed (?)

**Comment by habdelra** [comment]
when I get a chance, I'll checkout this branch to see if I can help with the TS linting issuing you are having

File: types/web3-provider-engine/index.d.ts (hunk 2 above)
Comment (habdelra) [reply types/web3-provider-engine/index.d.ts:1]: or wait never mind--it looks like this is not necessarily a compile artifact, and that you are intentionally including this to address the type errors you are running into.

File: packages/wc-provider/package.json (hunk 1 above)
Comment (Aierie) [reply packages/wc-provider/package.json:8]: Yes, this is LGPL-3.0. I wasn't sure if we wanted to handle this in the same PR - or if it should be in the same codebase?

Comment (habdelra) [reply packages/wc-provider/package.json:8]: it it's just a matter of some boilerplate text we need to include i think it's fine to do here. I vaguely recall @ef4 mentioning that segregating GPL in a mono repo into its own workspace was a thing that other projects have done.

**Comment by Aierie** [comment]
https://github.com/cardstack/cardstack/pull/2546 has tests that are currently failing for usage of the provider with Ethers.

**Comment by habdelra** [comment]
I was looking at the typings for @types/web3-provider-engine and it's pretty bleak. I think you'll need to keep the `// @ts-ignore` since no one has made typings for the modules that you need to import :-(

**Comment by habdelra** [comment]
@Aierie just an FYI: I test this branch in my local discord server, and the wallet connect logic worked just fine. I was able to perform a wallet connect thru the discord bot and receive a prepaid card. I you would like I can send you an invite to my discord server and share with you my .env files for this so that you can test it out too.

**Review by habdelra — APPROVED** [review]
Aside form teh GPL license, i think this is good to go. passes my local tests

**Comment by habdelra** [comment]
@Aierie actually i do see this error whern trying to connect via wallet connect in the server logs:
```(node:8438) UnhandledPromiseRejectionWarning: Error: No RPC Url available for chainId: 100
    at WalletConnectProvider.updateRpcUrl (/Users/hassan/codez/cardstack/packages/hub/dist/hub.js:682905:32)
    at WalletConnectProvider.updateState (/Users/hassan/codez/cardstack/packages/hub/dist/hub.js:682895:14)
    at Object.callback (/Users/hassan/codez/cardstack/packages/hub/dist/hub.js:682830:34)
    at /Users/hassan/codez/cardstack/packages/hub/dist/hub.js:234376:30
    at Array.forEach (<anonymous>)
    at EventManager.trigger (/Users/hassan/codez/cardstack/packages/hub/dist/hub.js:234370:23)
    at WalletConnect._handleSessionResponse (/Users/hassan/codez/cardstack/packages/hub/dist/hub.js:235079:40)
    at Object.callback (/Users/hassan/codez/cardstack/packages/hub/dist/hub.js:235150:22)
    at /Users/hassan/codez/cardstack/packages/hub/dist/hub.js:234376:30
    at Array.forEach (<anonymous>)
    at EventManager.trigger (/Users/hassan/codez/cardstack/packages/hub/dist/hub.js:234370:23)
    at WalletConnect.<anonymous> (/Users/hassan/codez/cardstack/packages/hub/dist/hub.js:235133:36)
    at Generator.next (<anonymous>)
    at fulfilled (/Users/hassan/codez/cardstack/packages/hub/dist/hub.js:683150:58)
    at processTicksAndRejections (internal/process/task_queues.js:95:5)
(node:8438) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 11)
(node:8438) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```
it's not a fatal error though. and the prepaid card provisioning can occur despite this error. probably, though, it would be good to see if we can get to the bottom of this though

**Comment by habdelra** [comment]
in regards to that error, i think that is actually on me. i just realized that I had the wallet set to xdai (but staging cards use sokol). i think that is just user error

File: packages/wc-provider/package.json (hunk 1 above)
Comment (Aierie) [reply packages/wc-provider/package.json:8]: WalletConnect has changed to Apache-2.0 as of `1.7.0` so I've changed the files to refer to that in https://github.com/cardstack/cardstack/pull/2544/commits/456a54daf76ff056147c62ce7606b56b150acefe

**Comment by habdelra** [comment]
just double checking: we still need to point to the forked `ethereum-test-network`, right?

**Comment by Aierie** [comment]
> just double checking: we still need to point to the forked `ethereum-test-network`, right?

Yep!

**Pull request merged**